smartsnippets/admin.py

```diff
@@ -79,6 +80,23 @@ def clean_template_path(self):
             raise ValidationError(e)
         return path
 
+    def clean(self):
+        clean_result = super(SnippetForm, self).clean()
+        self.validate_unique_variable_names()
+        return clean_result
+
+    def validate_unique_variable_names(self):
+        """ Validates name uniqueness over all variable inlines. """
+        unique_variable_names = set()
+        all_variable_names = [clean_variable_name(value)
+                              for key, value in self.data.dict().iteritems()
+                              if re.match(r"variables[0-9-]*name", key)]
+        for variable_name in all_variable_names:
+            if variable_name in unique_variable_names:
+                raise ValidationError(
```

Comment:
(Optional - fairly minor)

What if you have a snippet that has multiple variables duplicated?
Your error only shows the first duplication.
The user fixes that issue, but the save fails again with the next duplication.

Would it not be better to print an error message containing all duplications?